enterprise/dev/ci/gen-pipeline.go

```diff
@@ -3,22 +3,63 @@
 package main
 
 import (
+	"flag"
+	"fmt"
+	"io"
 	"os"
+	"strings"
 	"time"
 
+	"github.com/sourcegraph/sourcegraph/enterprise/dev/ci/internal/buildkite"
 	"github.com/sourcegraph/sourcegraph/enterprise/dev/ci/internal/ci"
 )
 
+var preview bool
+
+func init() {
+	flag.BoolVar(&preview, "preview", false, "Preview the pipeline steps")
+}
+
 func main() {
+	flag.Parse()
+
 	config := ci.NewConfig(time.Now())
 
 	pipeline, err := ci.GeneratePipeline(config)
 	if err != nil {
 		panic(err)
 	}
 
+	if preview {
+		previewPipeline(os.Stdout, config, pipeline)
+		return
+	}
+
 	_, err = pipeline.WriteTo(os.Stdout)
 	if err != nil {
 		panic(err)
 	}
 }
+
+func previewPipeline(w io.Writer, c ci.Config, bk *buildkite.Pipeline) {
+	fmt.Fprintf(w, "Detected run type: %s\n", c.RunType.String())
+	fmt.Fprintf(w, "Detected Changed files (%d):\n", len(c.ChangedFiles))
+	for _, f := range c.ChangedFiles {
+		fmt.Fprintf(w, "\t%s\n", f)
+	}
+	fmt.Fprintln(w, "Detected changes:")
+	fmt.Fprintf(w, "\tAffects Client: %t\n", c.ChangedFiles.AffectsClient())
+	fmt.Fprintf(w, "\tAffects Go: %t\n", c.ChangedFiles.AffectsGo())
+	fmt.Fprintf(w, "\tAffects DockerFile: %t\n", c.ChangedFiles.AffectsDockerfiles())
```

Comment:
```suggestion
	fmt.Fprintf(w, "\tAffects Dockerfiles: %t\n", c.ChangedFiles.AffectsDockerfiles())
```

might be confusing to see something not here 🤔

Reply:
Yes, I thought about this too and I think we're good for now, but on the long run,  I agree being able to grab the whole set of `Affects*` is going to be better.